### ISSUE TITLE
`sendMediaGroup` throws retrofit2 exception

### ISSUE THREAD

**Issue by bash-spbu** [kotlin-telegram-bot/kotlin-telegram-bot#45]
I get the following error when invoke `Bot#sendMediaGroup` method :
```traceback
Exception in thread "pool-1-thread-1" java.lang.IllegalArgumentException: Parameter type must not include a type variable or wildcard: java.util.List<? extends me.ivmg.telegram.entities.inputmedia.InputMedia> (parameter #2)
    for method ApiService.sendMediaGroup
	at retrofit2.ServiceMethod$Builder.methodError(ServiceMethod.java:752)
	at retrofit2.ServiceMethod$Builder.methodError(ServiceMethod.java:743)
	at retrofit2.ServiceMethod$Builder.parameterError(ServiceMethod.java:761)
	at retrofit2.ServiceMethod$Builder.build(ServiceMethod.java:195)
	at retrofit2.Retrofit.loadServiceMethod(Retrofit.java:170)
	at retrofit2.Retrofit$1.invoke(Retrofit.java:147)
	at com.sun.proxy.$Proxy0.sendMediaGroup(Unknown Source)
	at me.ivmg.telegram.network.ApiClient.sendMediaGroup(ApiClient.kt:471)
	at me.ivmg.telegram.Bot.sendMediaGroup(Bot.kt:436)
	at me.ivmg.telegram.Bot.sendMediaGroup$default(Bot.kt:435)
```

In my opinion problem is here:  
https://github.com/kotlin-telegram-bot/kotlin-telegram-bot/blob/ab798c96a3a6960a61cc9112842af636b1344543/telegram/src/main/kotlin/me/ivmg/telegram/network/ApiService.kt#L256

It is because Kotlin's List is covariant and to support it Kotlin's compiler generates Java's wildcards. So indeed it's something like 
```java
@Field("media") List<? extends InputMedia> media,
```

As I think the *possible fixes* for it is either to add `@JvmSuppressWildcards` annotation to this parameter
```kotlin
@Field("media") media: List<@JvmSuppressWildcards InputMedia>,
```

either try to use `Array<InputMedia>` or `MutableList<InputMedia>` instead of `List<InputMedia>`.

*Additional info:*
- https://kotlinlang.org/docs/reference/java-to-kotlin-interop.html#variant-generics
- https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-suppress-wildcards/index.html
- https://stackoverflow.com/questions/45409461/the-generic-parameters-any-of-kotlin-are-converted-to-wildcards

**Comment by bash-spbu** [comment]
And moreover, looks like this method needs `@Multipart` encoding, because now media is encoded in urlencoded form instead of being JSON-encoded :c

**Comment by vjgarciag96** [comment]
Hi @bash-spbu !! You're right. There are several errors on this method. 

1. Using `List<MediaType>` as parameter in the method throws the exception you commented.
2. `MediaType` is not the right type for `media` parameter because it has four childs: `InputMediaAudio`, `InputMediaDocument`, `InputMediaVideo` and `InputMediaPhoto`. However, the documentation says that the `sendMediaGroup` operation only accepts `InputMediaPhoto` and `InputMediaVideo`.
3. The method should return an array of `Message` instead of a single message as mentioned in the documentation.
4. Finally, the last thing you commented. There are several ways to send media to a chat. It's possible to pass a `file_id`, an `url` or to use _multipart/form-data_.

To sum up, our API should provide the three ways of sending a media group to a chat. We'll try to add them in the near future, stay tuned.

And thank you so much for reporting this problem! The info you provided was really useful.

**Comment by vjgarciag96** [comment]
@bash-spbu fixed by #60! Closing it
